DQMOffline/Lumi/plugins/ZCounting.cc

```diff
@@ -388,8 +440,222 @@ void ZCounting::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
+          eleProbe = ele;
+          break;
+        }
+      }
+
+      // Assign final probe 4-vector
+      TLorentzVector vProbe(0.,0.,0.,0.);
+      if(eleProbe.isNonnull()){
+        vProbe.SetPtEtaPhiM( eleProbe->pt(), eleProbe->eta(), eleProbe->phi(), ELECTRON_MASS);
+      } else {
+        double pt = sc->energy() * sqrt( 1 - pow(tanh(sc->eta()),2) );
+        vProbe.SetPtEtaPhiM( pt, sc->eta(), sc->phi(), ELECTRON_MASS);
+      }
+
+      // Probe Selection
+      double probe_pt = vProbe.Pt();
+      double probe_abseta = fabs(sc->eta());
+      bool probe_is_valid_probe = ele_probe_selection(probe_pt, probe_abseta);
+      if( !probe_is_valid_probe ) continue;
+
+      // Good Probe found!
+
+      // Require good Z
+      TLorentzVector vDilep = vTag + vProbe;
+
+      if((vDilep.M()<ELE_MASS_CUT_LOW) || (vDilep.M()>ELE_MASS_CUT_HIGH)) continue;
+      if(eleProbe.isNonnull() and (eleProbe->charge() != - el1->charge())) continue;
+
+      // Good Z found!
+
+      long ls = iEvent.luminosityBlock();
+      bool probe_pass_trigger = isElectronTriggerObj(*fTrigger, TriggerTools::matchHLT(vProbe.Eta(), vProbe.Phi(), fTrigger->fRecords, *hTrgEvt));
+      bool probe_pass_id = eleProbe.isNonnull() and EleID_.passID(eleProbe,beamspotHandle,conversionsHandle);
+
+      //// Fill for yields
+      bool probe_is_forward = probe_abseta > ELE_ETA_CRACK_LOW;
+      bool tag_is_forward = tag_abseta > ELE_ETA_CRACK_LOW;
+
+      if(probe_pass_id) {
+        if(probe_is_forward and tag_is_forward) {
+          h_ee_yield_Z_eeee -> Fill(ls);
+        } else if(!probe_is_forward and !tag_is_forward) {
+          h_ee_yield_Z_ebeb -> Fill(ls);
+        } else {
+          h_ee_yield_Z_ebee -> Fill(ls);
+        }
+      }
+
+      if(!tag_is_valid_tag) continue;
+
+      /// Fill for ID efficiency
+      if(probe_pass_id) {
+        if(probe_is_forward) {
+          h_ee_mass_id_pass_forward->Fill(ls, vDilep.M());
+        } else {
+          h_ee_mass_id_pass_central->Fill(ls, vDilep.M());
+        }
+      } else {
+          if(probe_is_forward) {
+            h_ee_mass_id_fail_forward->Fill(ls, vDilep.M());
+          } else {
+            h_ee_mass_id_fail_central->Fill(ls, vDilep.M());
+          }
+      }
+
+      /// Fill for HLT efficiency
+      if(probe_pass_id and probe_pass_trigger) {
+        if(probe_is_forward) {
+          h_ee_mass_HLT_pass_forward->Fill(ls, vDilep.M());
+        } else {
+          h_ee_mass_HLT_pass_central->Fill(ls, vDilep.M());
+        }
+      } else if (probe_pass_id) {
+        if(probe_is_forward) {
+          h_ee_mass_HLT_fail_forward->Fill(ls, vDilep.M());
+        } else {
+          h_ee_mass_HLT_fail_central->Fill(ls, vDilep.M());
+        }
+      }
+    } // End of probe loop
+  }//End of tag loop
+
 }
 
+bool ZCounting::ele_probe_selection(double pt, double abseta){
+  if(pt < ELE_PT_CUT_PROBE) return false;
+  else if(abseta > ELE_ETA_CUT_PROBE) return false;
```

Comment:
these elses are not needed. (as the function exits if true)

Reply:
Removed elses.